src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = { ...state };
+  const stateChangesArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (true) {
+      case actions[i].type === 'addProperties':
+        const { extraData } = actions[i];
+
+        Object.assign(currentState, extraData);
+        stateChangesArr[i] = Object.assign({}, currentState);
```

Comment:
this expression repeats three times maybe it is better to use if? you will able to write this statement only once at the end of loop